### PR TITLE
Add OpenTracing stuff

### PR DESCRIPTION
**Pull request by tomwilkie**
So we can visualise query execution, to help with #132

### CODE REVIEW

**Comment by tomwilkie** [comment]
(I'm just pushing the changes to the scope vendored code back to scope)

**Comment by tomwilkie** [comment]
Pay attention to the bits in:
- cmd/cortex/main.go
- distributor/distributor.go
- util/middleware/auth.go
- util/http.go

Have tested extensively locally.

File: cmd/cortex/main.go [hunk 1]
```diff
@@ -139,6 +143,12 @@ func main() {
 		prometheus.MustRegister(resourceWatcher)
 	}
 
+	tracer, err := loki.NewTracer(loki.DefaultConfig)
```
Comment (jml) [inline cmd/cortex/main.go]: There should be command-line options for scrape intervals & timeouts.

File: chunk/chunk_store.go [hunk 2]
```diff
@@ -417,7 +417,7 @@ func (c *AWSStore) Get(ctx context.Context, from, through model.Time, matchers .
 	}
 
 	// TODO push ctx all the way through, so we can do cancellation (eventually!)
```
Comment (jml) [inline chunk/chunk_store.go]: Are we doing this TODO now?

File: cmd/cortex/main.go [hunk 3]
```diff
@@ -12,7 +12,11 @@ import (
 	"time"
 
 	"github.com/gorilla/mux"
+	"github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc"
 	"github.com/mwitkow/go-grpc-middleware"
+	"github.com/opentracing-contrib/go-stdlib/nethttp"
+	"github.com/opentracing/opentracing-go"
+	"github.com/tomwilkie/loki/client"
```
Comment (tomwilkie) [inline cmd/cortex/main.go]: This is a personal project right now.

File: chunk/chunk_store.go (hunk 2 above)
Comment (tomwilkie) [inline chunk/chunk_store.go]: Yup, done.

File: cmd/cortex/main.go (hunk 1 above)
Comment (jml) [inline cmd/cortex/main.go]: In Prometheus, they are defined in the configuration file. Here, we're using the defaults in Loki with no way of customizing them.

Comment (tomwilkie) [inline cmd/cortex/main.go]: I've literrally copied the exact way prometheus does this: https://github.com/tomwilkie/loki/blob/master/pkg/config/config.go

Are we talking cross purposes?

**Pull request closed without merging**